server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -444,7 +604,10 @@ public void writeTo(StreamOutput out) throws IOException {
             o.writeString(role.roleNameAbbreviation());
             o.writeBoolean(role.canContainData());
         });
-        Version.writeVersion(version, out);
+        Version.writeVersion(nodeVersion, out);
+        if (out.getTransportVersion().onOrAfter(TRANSPORT_VERSION_VERSION)) {
```

Comment:
I *think* this recursive reference to transport version will Just Work, but if someone could confirm that would help